compiler/ilgen/MethodBuilder.cpp

```diff
@@ -522,14 +522,16 @@ MethodBuilder::DefineFunction(const char* const name,
       }   
    MB_REPLAY(");");
 
-   TR::ResolvedMethod *method = new (PERSISTENT_NEW) TR::ResolvedMethod((char*)fileName,
-                                                                        (char*)lineNumber,
-                                                                        (char*)name,
-                                                                        numParms,
-                                                                        parmTypes,
-                                                                        returnType,
-                                                                        entryPoint,
-                                                                        0);
+   // Store ResolvedMethod in the heap region of the TypeDictionary memory segment
+   TR_Memory *memory = typeDictionary()->trMemory();
```

Comment:
I don't see any issue with doing that. @lmaisons can you think of any reason why this might not work out?